eventuals/notification.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "eventuals/lock.h"
+#include "eventuals/then.h"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace eventuals {
+
+////////////////////////////////////////////////////////////////////////
+
+// This header file defines a 'Notification' abstraction, which allows
+// an eventual to be able to wait for a single occurrence of a single
+// event. Said another way, the 'Notification' object maintains a
+// private boolean "notified" that transitions to 'true' at most once.
+//
+// This is similar to 'absl::Notification'
+// (https://abseil.io/docs/cpp/guides/synchronization#notification).
+//
+// You can call 'Wait()' on a 'Notification' to wait until the
+// "notified" state is 'true' and 'Notify()' to set the notification's
+// "notified" state to 'true' and notify all waiting eventuals that
+// the event has occurred. This method should only be called once and
+// will do nothing on repeated calls.
```

Comment:
WDYT about mimicking `absl`'s behavior of disallowing repeated `Notify()` calls, since they're likely due to buggy code?

`absl` does this by crashing in non-opt builds (to make it super quick and easy to notice and find your buggy usage), but not crashing in opt builds (where you might not want bugs to crash your server): https://github.com/abseil/abseil-cpp/blob/149ab40581d33febf3f8bddb52505c9ee9edd838/absl/synchronization/notification.cc#L30-L37

IMHO ~`CHECK` failures on errors (at least optionally / by default) make it much easier to get your code to a bug-free state.